### PR TITLE
[ts2pant-body-lowering-completeness-rd2] Patch 1: Commit corpus diagnostic and add round-2 fixture stubs

### PR DESCRIPTION
**Pull request by subsetpark**
## Patch 1: Commit corpus diagnostic and add round-2 fixture stubs

- Commit the corpus diagnostic script reconstructed during planning; keep it read-only and free of test-suite side effects.
- Add one constructs fixture file that isolates each targeted shape and each preserved-rejection shape.
- Add skipped unit stubs for nested pure block-return helper behavior, branch/switch/callback consumers, and record-return conditional lowering.
- Snapshot the fixture file before behavior changes, showing the current UNSUPPORTED lines.

## Changes
- Commit the corpus diagnostic script reconstructed during planning; keep it read-only and free of test-suite side effects.
- Add one constructs fixture file that isolates each targeted shape and each preserved-rejection shape.
- Add skipped unit stubs for nested pure block-return helper behavior, branch/switch/callback consumers, and record-return conditional lowering.
- Snapshot the fixture file before behavior changes, showing the current UNSUPPORTED lines.

## Files to Modify
- tools/ts2pant/scripts/corpus-diag.mts (create): Read-only dogfood diagnostic that scans top-level ts2pant src functions, emits each document, and ranks normalized UNSUPPORTED buckets.
- tools/ts2pant/tests/fixtures/constructs/expressions-body-lowering-rd2.ts (create): Positive and negative fixtures for nested branch blocks, terminal if/else branch blocks, switch nested block clauses, callback block bodies, record-return conditionals, and deferred local accumulator sequencing.
- tools/ts2pant/tests/translate-body.test.mts (modify): Add skipped unit-test stubs for the shared helper and each behavior patch, marked PENDING with owning patch numbers.
- tools/ts2pant/tests/constructs.test.mts.snapshot (modify): Snapshot the new fixtures in their current rejected state.

## Gameplan Specification

```
module BODY_LOWERING_COMPLETENESS_RD2.

Body.
Block.
Arm.
Clause.
Callback.
RecordBranchSet.

supported-pure-block? b: Block => Bool.
block-lowered? b: Block => Bool.
nested-block-arm? a: Arm => Bool.
arm-lowered? a: Arm => Bool.
nested-switch-clause? c: Clause => Bool.
clause-lowered? c: Clause => Bool.
nested-callback-block? cb: Callback => Bool.
callback-lowered? cb: Callback => Bool.
same-field-set? r: RecordBranchSet => Bool.
fieldwise-cond-lowered? r: RecordBranchSet => Bool.
unsupported-shape? b: Body => Bool.
rejected? b: Body => Bool.
typechecks? b: Body => Bool.

---

all b: Block | supported-pure-block? b -> block-lowered? b.
all a: Arm | nested-block-arm? a -> arm-lowered? a.
all c: Clause | nested-switch-clause? c -> clause-lowered? c.
all cb: Callback | nested-callback-block? cb -> callback-lowered? cb.
all r: RecordBranchSet | same-field-set? r -> fieldwise-cond-lowered? r.
all b: Body | unsupported-shape? b -> rejected? b.
all b: Body | typechecks? b.
```

## Patch Specification

```
module BODY_LOWERING_COMPLETENESS_RD2_PATCH_1.

Fixture.
positive? f: Fixture => Bool.
negative? f: Fixture => Bool.
snapshotted? f: Fixture => Bool.

---

all f: Fixture | positive? f or negative? f -> snapshotted? f.
```

## Implementation Notes

- The corpus diagnostic intentionally uses the production pipeline (`buildPantDocument` + `emitDocument`) rather than test helpers, so it stays read-only and does not invoke wasm typechecking or mutate snapshots. It supports `--summary-only`; without that flag it prints every emitted document before the bucket summary.
- Bucket normalization only strips the leading `<function> — ` prefix and numeric literals. I initially considered masking kebab-case names, but that hid important diagnostic labels like `if-with-return`, so those are preserved for before/after comparisons.
- The new construct fixture mixes future-positive shapes with preserved-rejection shapes. All entries are currently snapshotted as `UNSUPPORTED`; dependent behavior patches should update only the snapshots for the shapes they intentionally admit and leave local accumulator sequencing, non-literal switch labels, fall-through, effectful consts, and mismatched record fields rejected.
- The skipped unit tests are deliberately placed near existing translate-body unsupported/edge-case tests and named with owning patch numbers (`PENDING Patch 2` through `PENDING Patch 5`) to make unskipping ownership explicit.

Spec/Evidence Mapping:
- `positive? f or negative? f -> snapshotted? f`: implemented by `tools/ts2pant/tests/fixtures/constructs/expressions-body-lowering-rd2.ts` plus generated entries in `tools/ts2pant/tests/constructs.test.mts.snapshot`.
- Required context consulted: existing construct harness/snapshots, `translate-body.test.mts`, existing round-1 block/switch/record fixtures, and the gameplan corpus diagnostic context.
- Evidence: `just ts2pant-test-update-snapshots` passed with 1500 pass / 7 skipped / 0 fail; `NODE_OPTIONS=--max-old-space-size=6144 npx tsx scripts/corpus-diag.mts --summary-only` passed and reported the expected baseline of 806 functions, 163 clean, and 64 `if-with-return block must contain only const bindings followed by a return`.

### CODE REVIEW

File: tools/ts2pant/scripts/corpus-diag.mts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+let clean = 0;
+let unsupported = 0;
+let errored = 0;
+
+for (const result of results) {
+  if (result.error !== undefined) {
+    errored += 1;
+    const bucket = `ERROR: ${result.error.split("\n", 1)[0]}`;
+    buckets.set(bucket, (buckets.get(bucket) ?? 0) + 1);
+    continue;
```
[the remark this answers is not in the document]
Comment (subsetpark) [reply tools/ts2pant/scripts/corpus-diag.mts:125]: Addressed in commit 06760a9: corpus diagnostic errors now go into a separate errorBuckets map and print under an ERROR buckets section, leaving the UNSUPPORTED buckets distribution clean for before/after comparisons. Verified with NODE_OPTIONS=--max-old-space-size=6144 npx tsx scripts/corpus-diag.mts --summary-only.
